Assets/MixedRealityToolkit/_Core/Services/SpatialAwarenessSystem/MixedRealitySpatialAwarenessSystem.cs

```diff
@@ -0,0 +1,289 @@
+using Microsoft.MixedReality.Toolkit.Core.Interfaces.SpatialAwarenessSystem.Handlers;
+using System.Collections.Generic;
+using UnityEngine;
+using UnityEngine.EventSystems;
+
+namespace Microsoft.MixedReality.Toolkit.Core.Services.SpatialAwarenessSystem
+{
+    /// <summary>
+    /// Class providing the default implementation of the <see cref="IMixedRealitySpatialAwarenessSystem"/> interface.
+    /// </summary>
+    public class MixedRealitySpatialAwarenessSystem : BaseEventSystem, IMixedRealitySpatialAwarenessSystem
+    {
+        private GameObject spatialAwarenessParent = null;
+
+        /// <summary>
+        /// Parent <see cref="GameObject"/> which will encapsulate all of the spatial awareness system created scene objects.
+        /// </summary>
+        public GameObject SpatialAwarenessRootParent => spatialAwarenessParent != null ? spatialAwarenessParent : (spatialAwarenessParent = CreateSpatialAwarenessParent);
+
+        /// <summary>
+        /// Creates the parent for spatial awareness objects so that the scene hierarchy does not get overly cluttered.
+        /// </summary>
+        /// <returns>
+        /// The <see cref="GameObject"/> to which spatial awareness created objects will be parented.
+        /// </returns>
+        private GameObject CreateSpatialAwarenessParent => new GameObject("Spatial Awareness System");
+
+        private GameObject meshParent = null;
+
+        /// <inheritdoc />
+        public GameObject SpatialMeshesParent => meshParent != null ? meshParent : (meshParent = CreateSecondGenerationParent("Meshes"));
+
+        private GameObject surfaceParent = null;
+
+        /// <inheritdoc />
+        public GameObject SurfacesParent => surfaceParent != null ? surfaceParent : (surfaceParent = CreateSecondGenerationParent("Surfaces"));
+
+        /// <summary>
+        /// Creates a parent that is a child of the Spatial Awareness System parent so that the scene hierarchy does not get overly cluttered.
+        /// </summary>
+        /// <returns>
+        /// The <see cref="GameObject"/> to which spatial awareness objects will be parented.
+        /// </returns>
+        private GameObject CreateSecondGenerationParent(string name)
+        {
+            var secondGeneration = new GameObject(name);
+
+            secondGeneration.transform.parent = SpatialAwarenessRootParent.transform;
+
+            return secondGeneration;
+        }
+
+        #region IMixedRealitySpatialAwarenessSystem Implementation
+
+        /// <inheritdoc />
+        public HashSet<IMixedRealitySpatialObserverDataProvider> DetectedSpatialObservers { get; } = new HashSet<IMixedRealitySpatialObserverDataProvider>();
+
+        /// <inheritdoc />
+        public bool IsObserverRunning(IMixedRealitySpatialObserverDataProvider observer)
+        {
+            return false;
+        }
+
+        /// <inheritdoc />
+        public uint GenerateNewObserverId()
```

Comment:
Should this be public? If it needs to be so that it can be called by observers, it should be documented as not being for application use.

Reply:
We do the same thing in the input system to request a new unique source id for controllers. Same idea applies here.

Reply:
I don't understand. It should be used for custom implementations, which is why it's in the interface.

Reply:
Here's an example comment:

```
        /// <remarks>
        /// This method is to be called by implementations of the <see cref="IMixedRealitySpatialObserver"/> interface, and not by application code.
        /// </remarks>
```

Reply:
Sure.